templates/terraform/update_mask.erb

```diff
@@ -1,22 +1,39 @@
-updateMask := []string{}
-<% update_body_properties.each do |prop| -%>
-<%# UpdateMask documentation is not not obvious about which fields are supported or
+<%# Template for code adding update mask query parameter to update URL.
+
+   UpdateMask documentation is not not obvious about which fields are supported or
    how deeply nesting is supported.  For instance, if we change the field foo.bar.baz,
    it seems that *sometimes*, 'foo' is a valid value.  Other times, it needs to be
    'foo.bar', and other times 'foo.bar.baz'. If the defaults don't work for you,
-   You can customize the exact list of fields that are passed for a property
+   you can customize the exact list of fields that are passed for a property
    using `update_mask_fields`.
--#%>
-if d.HasChange("<%= prop.name.underscore -%>") {
+-%>
+updateMask := []string{}
+<% update_body_properties.each do |prop| -%>
 <%
 mask = prop.api_name
 if prop.update_mask_fields
   mask = prop.update_mask_fields.join(',')
 end
 -%>
+<% if prop.flatten_object -%>
+<% prop.properties.each do |nested| %>
```

Comment:
Can you add some spaces for indentation to make it extra-clear when we're in the if statement / loop?
Suggestion for this line:
```suggestion
<%   prop.properties.each do |nested| %>
```